src/Protocol/ActivityPub/Receiver.php

```diff
@@ -2025,4 +2027,137 @@ private static function hasArrived(string $id): bool
+					$type = self::TARGET_BCC;
+					break;
+			}
+			$receiver_list = JsonLD::fetchElementArray($object, $element, '@id');
+			if (empty($receiver_list)) {
+				continue;
+			}
+
+			foreach ($receiver_list as $receiver) {
+				if ($receiver == self::PUBLIC_COLLECTION) {
+					$targets[self::TARGET_GLOBAL] = ($element == 'as:to');
+					continue;
+				}
+
+				if ($receiver == $followers) {
+					$targets[self::TARGET_FOLLOWER] = true;
+					continue;
+				}
+				$targets[$type][] = Contact::getIdForURL($receiver);
+			}
+		}
+		return $targets;
+	}
+
+	private static function routeC2SActivities(string $type, string $object_type, string $object_id, int $uid, array $application, array $ldactivity)
+	{
+		switch ($type) {
+			case 'as:Create':
+				if (in_array($object_type, self::CONTENT_TYPES)) {
+					self::createContent($uid, $application, $ldactivity);
+				}
+				break;
+			case 'as:Update':
+				if (in_array($object_type, self::CONTENT_TYPES) && !empty($object_id)) {
+					self::updateContent($uid, $object_id, $application, $ldactivity);
+				}
+				break;
+			case 'as:Follow':
+				if (in_array($object_type, self::ACCOUNT_TYPES) && !empty($object_id)) {
+					self::followAccount($uid, $object_id, $ldactivity);
+				}
+				break;
+		}
+	}
+
+	private static function createContent(int $uid, array $application, array $ldactivity)
+	{
+		$object_data = self::processObject($ldactivity['as:object'], true);
+		$item = Processor::processC2SContent($object_data, $application, $uid);
+		Logger::debug('Got data', ['item' => $item, 'object' => $object_data]);
+
+		$id = Item::insert($item, true);
+		if (!empty($id)) {
+			$item = Post::selectFirst(['uri-id'], ['id' => $id]);
+			if (!empty($item['uri-id'])) {
+				System::jsonExit(Transmitter::createActivityFromItem($id));
+			}
+		}
+	}
+
+	private static function updateContent(int $uid, string $object_id, array $application, array $ldactivity)
+	{
+		$id = Item::fetchByLink($object_id, $uid);
+		$original_post = Post::selectFirst(['uri-id'], ['uid' => $uid, 'origin' => true, 'id' => $id]);
+		if (empty($original_post)) {
+			Logger::debug('Item not found or does not belong to the user', ['id' => $id, 'uid' => $uid, 'object_id' => $object_id, 'activity' => $ldactivity]);
+			return;
+		}
+
+		$object_data = self::processObject($ldactivity['as:object'], true);
+		$item = Processor::processC2SContent($object_data, $application, $uid);
+		if (empty($item['title']) && empty($item['body'])) {
+			Logger::debug('Empty body and title', ['id' => $id, 'uid' => $uid, 'object_id' => $object_id, 'activity' => $ldactivity]);
+			return;
+		}
+		$post = ['title' => $item['title'], 'body' => $item['body']];
+		Logger::debug('Got data', ['id' => $id, 'uid' => $uid, 'item' => $post]);
+		Item::update($post, ['id' => $id]);
+		Item::updateDisplayCache($original_post['uri-id']);
+
+		System::jsonExit(Transmitter::createActivityFromItem($id));
+	}
+
+	private static function followAccount($uid, $object_id, $ldactivity)
```

Comment:
empty hull? :)

Reply:
I have to figure out the expected return value since I don't know any documentation about it.